theano/gof/destroyhandler.py

```diff
@@ -937,8 +944,11 @@ def validate(self, fgraph):
 
         """
         if self.destroyers:
+            if config.disbale_cycle_detection and self.fail_validate:
+                self.fail_validate = False
+                # raise self.fail_validate
+                InconsistencyError("error")
```

Comment:
I think you want that here:
err = self.fail_validate
self.fail_validate = False
raise err